kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py

```diff
@@ -16,12 +27,448 @@ class KafkaCheck(AgentCheck):
+        are unspecified for a topic listed in the config, offsets are fetched for all the partitions within that topic.
+        """
+        coordinator_id = self._kafka_client._find_coordinator_id_process_response(response)
+        topics = self._consumer_groups[consumer_group]
+        if not topics:  # TODO will topics be None or empty list/set if unspecified??
+            topic_partitions = None  # None signals to fetch all known offsets for the consumer group
+        else:
+            topic_partitions = []  # transform [("t1", [1, 2])] into [TopicPartition("t1", 1), TopicPartition("t1", 2)]
+            for topic, partitions in topics.items():
+                if partitions is None:  # If partitions aren't specified, fetch all partitions in the topic
+                    partitions = self._kafka_client._client.cluster.partitions_for_topic(topic)
+                topic_partitions.extend([TopicPartition(topic, p) for p in partitions])
+        single_group_offsets_future = self._kafka_client._list_consumer_group_offsets_send_request(
+            group_id=consumer_group, group_coordinator_id=coordinator_id, partitions=topic_partitions
+        )
+        single_group_offsets_future.add_callback(self._single_group_offsets_callback, consumer_group)
+        self._consumer_futures.append(single_group_offsets_future)
+
+    def _single_group_offsets_callback(self, consumer_group, response):
+        """Callback that parses an OffsetFetchResponse and saves it to the consumer_offsets dict.
+
+        consumer_group must be manually passed in because it is not present in the response, but we need it in order to
+        associate these offsets to the proper consumer group.
+        """
+        single_group_offsets = self._kafka_client._list_consumer_group_offsets_process_response(response)
+        for (topic, partition), (offset, _metadata) in single_group_offsets.items():
+            # If the OffsetFetchRequest explicitly specified partitions, the offset could returned as -1, meaning there
+            # is no recorded offset for that partition... for example, if the partition doesn't exist in the cluster.
+            # So ignore it.
+            if offset == -1:
+                self._kafka_client._client.cluster.request_update()  # force metadata update on next poll()
+                continue
+            key = (consumer_group, topic, partition)
+            self._consumer_offsets[key] = offset
+
+    # TODO since this is used to validate the config interface, ideally this would be shared between new and legacy
+    # versions of the check to make sure the interface they accept doesn't diverge if someone updates one but forgets
+    # to update the other copy.
+    def _validate_listed_consumer_groups(self):
+        """Validate the user-specified list of consumer groups."""
+        # _consumer_groups example: {'consumer_group': {'topic': [0, 1]}}
+        # TODO can I allow consumer_groups to be a list rather than requiring a dict?
```

Comment:
No

Reply:
Why not? Couldn't this check accept both types?

Reply:
🤔 How would a list be defined in config?

Reply:
Same way any YAML list... it would only work if omitting the topics, but with this new code path topics are completely optional for the consumer groups:

    consumer_groups:
      <CONSUMER_GROUP_NAME_1>
      <CONSUMER_GROUP_NAME_2>

Reply:
Discussed on Slack, Datadog moving to preferring only one type of options... I can get behind that, so deleted this TODO note.